lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -381,6 +385,42 @@ private function generateNamespace(ClassMetadata $class)
         return strrev($parts[1]);
     }
 
+    /**
+     * Enums must have a use statement when used as public property defaults.
+     */
+    public function generateEnumUseStatements(ClassMetadata $class): string
+    {
+        if (PHP_VERSION_ID < 80100) {
+            return "\n";
+        }
+
+        $defaultProperties          = $class->getReflectionClass()->getDefaultProperties();
+        $lazyLoadedPublicProperties = $this->getLazyLoadedPublicPropertiesNames($class);
+        $enumClasses                = [];
+
+        foreach ($class->getReflectionClass()->getProperties(ReflectionProperty::IS_PUBLIC) as $property) {
+            $name = $property->getName();
+
+            if (! in_array($name, $lazyLoadedPublicProperties, true)) {
+                continue;
+            }
+
+            if (array_key_exists($name, $defaultProperties) && $defaultProperties[$name] instanceof BackedEnum) {
+                $enumClasses[] = get_class($defaultProperties[$name]);
+            }
+        }
+
+        return implode(
+            "\n",
+            array_map(
+                static function ($className) {
+                    return 'use ' . $className . ';';
```

Comment:
Will this work if it is a deep namespace? Ex var_export export class as 'Doctrine\Tests\Common\Proxy\YesNo' So should the "use" not only be "use Doctrine"?

Reply:
Yes, see the test assertion, https://github.com/doctrine/common/pull/955/files#diff-8bd50f2d6a0c9d08df7a09474c0ce3ddb0222b38b85984b1b723bfec5825e768R544

Reply:
I did not write what i meant clearly enough.

Try "new \Doctrine\Tests\Common\ProxyProxy\__CG__\Doctrine\Tests\Common\Proxy\Php81EnumPublicPropertyType()" and you will get "Error: Class "Doctrine\Tests\Common\ProxyProxy\__CG__\Doctrine\Tests\Common\Proxy\Doctrine\Tests\Common\Proxy\YesOrNo" not found"

I believe that we only have "use Doctrine" and not "use Doctrine\Tests\Common\Proxy\YesOrNo"

Reply:
A quick and dirty example which fix the issue could be:

$rootNamespaces = array_map(fn ($enumClass) => explode('\\', $enumClass)[0], $enumClasses);